objstorage/objstorageprovider/shared_obj_name.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+package objstorageprovider
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/pebble/internal/base"
+	"github.com/cockroachdb/pebble/objstorage"
+)
+
+// sharedObjectName returns the name of an object on shared storage.
+//
+// For sstables, the format is: <hash>/<creator-id>-<file-num>.sst
+// For example: 1a3f-00000000000000000002-000001.sst
+func sharedObjectName(meta objstorage.ObjectMetadata) string {
+	switch meta.FileType {
+	case base.FileTypeTable:
+		return fmt.Sprintf(
+			"%04x-%020d-%06d.sst",
```

Comment:
This seems to be out of sync with the comment? It's `hash-creatorID-filenum.sst` while the comment says it's `hash/creatorid-filenum.sst`. The latter is probably necessary for performance.